src/ucp/wireup/address.c

```diff
@@ -440,14 +645,86 @@ static uint64_t ucp_address_flags_from_iface_flags(uint64_t iface_flags)
     return iface_cap_flags;
 }
 
+static unsigned
+ucp_address_pack_iface_attr_v1(ucp_worker_h worker, void *ptr,
+                               const uct_iface_attr_t *iface_attr,
+                               unsigned atomic_flags)
+{
+    ucp_address_packed_iface_attr_t *packed = ptr;
+
+    packed->overhead       = iface_attr->overhead;
+    packed->bandwidth      = ucp_tl_iface_bandwidth(worker->context,
+                                                    &iface_attr->bandwidth);
+    packed->lat_ovh        = iface_attr->latency.c;
+    /* Pack prio, capability and atomic flags */
+    packed->prio_cap_flags =
+        (uint8_t)iface_attr->priority                      |
+        (atomic_flags << UCP_ADDRESS_FLAG_ATOMIC_SHIFT_V1) |
+        ucp_address_pack_flags(iface_attr->cap.flags,
+                               UCP_ADDRESS_IFACE_FLAGS, 8);
+    /* Keep only the bits defined by UCP_ADDRESS_IFACE_EVENT_FLAGS to shrink
+     * address. */
+    packed->prio_cap_flags |= ucp_address_pack_flags(
+            iface_attr->cap.event_flags, UCP_ADDRESS_IFACE_EVENT_FLAGS,
+            8 + ucs_popcount(UCP_ADDRESS_IFACE_FLAGS));
+
+    ucs_assert_always((ucs_popcount(UCP_ADDRESS_IFACE_FLAGS) +
+                ucs_popcount(UCP_ADDRESS_IFACE_EVENT_FLAGS)) <= 22);
+
+    return sizeof(*packed);
+}
+
+static unsigned
+ucp_address_pack_iface_attr_v2(ucp_worker_h worker, void *ptr,
+                               const uct_iface_attr_t *iface_attr,
+                               unsigned atomic_flags)
+{
+    unsigned seg_size = 0;
+    ucp_address_v2_packed_iface_attr_t *packed = ptr;
+    uint64_t addr_iface_flags;
+
+    packed->overhead  = UCS_FP8_PACK(OVERHEAD, iface_attr->overhead);
+    packed->bandwidth = UCS_FP8_PACK(BANDWIDTH,
+                                     ucp_tl_iface_bandwidth(worker->context,
+                                     &iface_attr->bandwidth));
+    packed->lat_ovh   = UCS_FP8_PACK(LATENCY, iface_attr->latency.c);
+    packed->prio      = (uint8_t)iface_attr->priority;
+    addr_iface_flags  = ucp_address_flags_from_iface_flags(
+                            iface_attr->cap.flags);
```

Comment:
we should define a specific set of flags in UCP address (as discussed) and not pack UCT iface flags with mask
this is because if we would want to enable additional flag from UCT which is not the last flag in UCT, it would not be possible (will shift all other flags)

Reply:
right, this is what is done here.
`ucp_address_flags_from_iface_flags` returns specific UCP flags based on supported UCT flags

Reply:
i think it's still returning flags from UCT layer